From 6366f2a51019f6f6a9b1dece906466550845c543 Mon Sep 17 00:00:00 2001
From: "Cui, Yuxin" <yuxin.cui@intel.com>
Date: Fri, 25 Feb 2022 15:29:20 +0800
Subject: [PATCH] Check android.permission.DUMP for suspend_control

Check android.permission.DUMP at the beginning of the dump function.

Tracked-On: OAM-100531
Signed-off-by: Cui, Yuxin <yuxin.cui@intel.com>
---
 suspend/1.0/default/SuspendControlService.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/suspend/1.0/default/SuspendControlService.cpp b/suspend/1.0/default/SuspendControlService.cpp
index 671f3dd..72cae2c 100644
--- a/suspend/1.0/default/SuspendControlService.cpp
+++ b/suspend/1.0/default/SuspendControlService.cpp
@@ -19,6 +19,7 @@
 #include <android-base/logging.h>
 #include <android-base/stringprintf.h>
 #include <signal.h>
+#include <binder/IServiceManager.h>
 
 #include "SystemSuspend.h"
 
@@ -118,6 +119,9 @@ static std::string dumpUsage() {
 }
 
 status_t SuspendControlService::dump(int fd, const Vector<String16>& args) {
+    if (checkCallingPermission(String16("android.permission.DUMP")) == false) {
+        return PERMISSION_DENIED;
+    }
     register_sig_handler();
 
     const auto suspendService = mSuspend.promote();
-- 
2.25.1

